connector/src/main/java/org/jboss/as/connector/subsystems/datasources/AbstractDataSourceService.java

```diff
@@ -488,16 +487,13 @@ protected org.jboss.jca.core.spi.security.SubjectFactory getSubjectFactory(
             // safe assertion because all parsers create Credential
             assert credential instanceof Credential;
             final String securityDomain = credential.getSecurityDomain();
-            if (((Credential) credential).isElytronEnabled()) {
-                try {
-                    return new ElytronSubjectFactory(authenticationContext.getOptionalValue(), new java.net.URI(jndiName));
-                } catch (URISyntaxException e) {
-                    throw ConnectorLogger.ROOT_LOGGER.cannotDeploy(e);
-                }
-            } else if (securityDomain == null || securityDomain.trim().equals("") || subjectFactory.getOptionalValue() == null) {
-                return null;
-            } else {
-                return new PicketBoxSubjectFactory(subjectFactory.getValue());
+            if (securityDomain != null) {
+                DS_DEPLOYER_LOGGER.legacySecurityNotSupported();
```

Comment:
This method produces an OperationFailedException but it is not thrown. I don't think throwing OFE from here is correct; some other exception type is needed.

Reply:
fixed